app/src/app/[lng]/authorize/AuthorizeForm.tsx

```diff
@@ -0,0 +1,187 @@
+      const params = new URLSearchParams(
+        state
+          ? { error, error_description, state }
+          : { error, error_description },
+      );
+      router.push(`${redirectUri}?${params}`);
+    }
+  }, [redirectUri, clientId, router, state, responseType]);
+
+  useEffect(() => {
+    if (
+      redirectUri &&
+      clientId &&
+      (!scope || scope.split(" ").some((s) => !["read", "write"].includes(s)))
+    ) {
+      const error = "invalid_scope";
+      const error_description = "Valid scopes: read, write";
+      const params = new URLSearchParams(
+        state
+          ? { error, error_description, state }
+          : { error, error_description },
+      );
+      router.push(`${redirectUri}?${params}`);
+    }
+  }, [redirectUri, clientId, router, state, scope]);
+
+  useEffect(() => {
+    if (
+      redirectUri &&
+      clientId &&
+      (!codeChallenge || !codeChallengeMethod || codeChallengeMethod !== "S256")
+    ) {
+      const error = "invalid_request";
+      const error_description = "PKCE required";
+      const params = new URLSearchParams(
+        state
+          ? { error, error_description, state }
+          : { error, error_description },
+      );
+      router.push(`${redirectUri}?${params}`);
+    }
+  }, [
+    redirectUri,
+    clientId,
+    router,
+    state,
+    codeChallenge,
+    codeChallengeMethod,
+  ]);
+
+  if (!clientId || !redirectUri) {
+    return null;
+  }
+
+  const handleAuthorize = async () => {
+    toaster.loading({
+      title: t("oauth-generating-authorization-code"),
+      type: "info",
+    });
+    const { data, error } = await generateCode({
+      clientId,
+      redirectUri,
+      codeChallenge,
+      scope,
+      csrfToken
+    });
+    toaster.dismiss();
+    if (!error) {
+      const code = data;
+      const params = new URLSearchParams(state ? { code, state } : { code });
+      router.push(`${redirectUri}?${params}`);
+    } else {
+      const errorDescription = error instanceof Error ? error.message : 'Unknown error occurred';
+      const params = new URLSearchParams({
+        error: 'server_error',
+        error_description: errorDescription,
+        ...(state && { state })
+      });
+      router.push(`${redirectUri}?${params}`);
+    }
+  };
+
+  const handleCancel = () => {
+    const error = "access_denied";
+    const params = new URLSearchParams(state ? { error, state } : { error });
+    router.push(`${redirectUri}?${params}`);
+  };
+
+  return (
+    <Suspense fallback={<Text>{t("loading")}</Text>}>
+      <Box maxW="md" mx="auto" mt={12} p={6} borderWidth={1} borderRadius="lg">
+        <VStack gap={4} textAlign="center">
+          <Heading size="lg">{t("oauth-authorize-heading")}</Heading>
+          <Text>{t("oauth-authorize-prompt", { clientName: (isClientLoading) ? t("oauth-unknown-app") : client?.name.en })}</Text>
+          <ButtonGroup>
+            <Button colorScheme="blue" onClick={handleAuthorize}>
```

Comment:
Would be good to use the CC theme instead so it adjusts to the user's selected theme